### PR TITLE
Fetching config from another agent only logs if that agent is enabled

### LINKED ISSUE #24177: If the System Probe Agent or Security Agent is disabled, inventory agent may be producing noisy logs.
Hi Team,

I've noticed a recent update where the Inventory Agent now requests metadata directly from each Datadog Agent, rather than relying on local metadata cache. 
This change, as detailed in PR #22922, means that Inventory Agents periodically fetch metadata from each Agent.

This approach is designed to retrieve metadata irrespective of an Agent's status (enabled/disabled). 
When an Agent is offline, the system fallbacks to using local metadata. 
However, this process outputs specific log entries that may need attention:

```
2024-03-28 10:36:14 UTC | CORE | INFO | (comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:196 in getCorrectConfig) | could not fetch system-probe process configuration: Could not reach system-probe: Get "http://localhost/config": dial unix /opt/datadog-agent/run/sysprobe.sock: connect: no such file or directory
2024-03-28 10:36:14 UTC | CORE | INFO | (comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:196 in getCorrectConfig) | could not fetch security-agent process configuration: Could not reach security-agent: Get "https://localhost:5010/agent/config": dial tcp 127.0.0.1:5010: connect: connection refused
Make sure the system-probe is running before requesting the runtime configuration and contact support if you continue having issues
Make sure the security-agent is running before requesting the runtime configuration and contact support if you continue having issues
```

The logs are output at every inventory check interval, contributing to noise and potential confusion, especially since they suggest operational actions ("Make sure the foo agent is running before requesting the runtime configuration and contact support if you continue having issues") that may not be relevant at the time.

Could we consider suppressing these logs or modifying the messages for clarity?

Thanks!

### PR DESCRIPTION
**Pull request by dustmop**
### What does this PR do?

When fetching a config from another agent, do not log about that agent being unreachable if it is disabled by the core agent configuration.

### Motivation

Fixes https://github.com/DataDog/datadog-agent/issues/24177

### CODE REVIEW

File: comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go [hunk 1]
```diff
@@ -232,85 +243,82 @@ func (ia *inventoryagent) fetchCoreAgentMetadata() {
...
 
 func (ia *inventoryagent) fetchSystemProbeMetadata() {
-	// If the system-probe configuration is not loaded we fallback on zero value for all metadata
-	getBoolSysProbe := func(_ string) bool { return false }
-	getIntSysProbe := func(_ string) int { return 0 }
-
+	var sysProbeConf configGetter
 	localSysProbeConf, isSet := ia.sysprobeConf.Get()
-	if isSet {
+	if isSet && ia.conf.GetBool("system_probe_config.enabled") {
```
Comment (paulcacheux) [inline comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:270]: You can have the system probe running even with this config disabled, especially if you have `runtime_security_config.enabled`

**Comment by GustavoCaso** [comment]
Where is the logging coming from? Could you provide an example?

Could we filter out the logs any other way? It feels that we are patching the logging issue at the wrong layer. Could we tweak the logging at the client level? If I understand correctly the [client logic](https://github.com/DataDog/datadog-agent/blob/039d28294c5cd6f06d7e5694cb542c8cecac66aa/pkg/config/fetcher/from_processes.go#L7) checks for the other agent ports, could we early return if the port is not configured?  Could we doing something on that layer than in the inventory agent component?

File: comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go [hunk 2]
```diff
@@ -232,85 +243,82 @@ func (ia *inventoryagent) fetchCoreAgentMetadata() {
 }
 
 func (ia *inventoryagent) fetchSecurityAgentMetadata() {
-	securityCfg := ia.getCorrectConfig("security-agent", ia.conf, fetchSecurityConfig, ia.conf)
+	securityCfg := ia.getCorrectConfig("security-agent", "runtime_security_config.enabled", ia.conf, fetchSecurityConfig)
```
[the remark this answers is not in the document]
Comment (dustmop) [reply comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go]: The initial approach I was using was incorrect. I have changed it to instead not log at all when the other agents can't be reached. As you pointed out, trying to detect whether another agent flavor is running just from the config doesn't work.

**Comment by dustmop** [comment]
I've updated the PR so that now it's just the removal of the specific log.info we want to quiet, plus some related cleanup. No longer checking config settings to decide whether or not to fetch config from other agents.

**Comment by pgimalac** [comment]
> No longer checking config settings to decide whether or not to fetch config from other agents.

Any particular reason for that ?
At least a debug log could help, and it would be included when running a flare

**Review by GustavoCaso — APPROVED** [review]
LGTM 🎉 

Would be interesting to link the GH issue in the PR description

**Comment by dustmop** [comment]
> Any particular reason for that ? At least a debug log could help, and it would be included when running a flare

There's no way to know whether the other agent isn't running at all because it is not enabled, or if it should be running but has some issue. Only the latter is an error, but we have no way of telling the difference.

File: releasenotes/notes/less-logging-fetch-config-40f70fd190b23720.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+fixes:
+  - |
+    Don't log when failing to fetch config of other agents
```
Comment (jhgilbert) [inline releasenotes/notes/less-logging-fetch-config-40f70fd190b23720.yaml:11]:
```suggestion
    Don't log when failing to fetch config of other Agents
```

**Review by jhgilbert — APPROVED** [review]
Approved with a minor suggestion, please ignore if you didn't mean "other Datadog Agents". Thanks!

**Comment by pgimalac** [comment]
> There's no way to know whether the other agent isn't running at all because it is not enabled, or if it should be running but has some issue. Only the latter is an error, but we have no way of telling the difference.

Yes I know what the issue is, I meant that having a debug log would hide it from customers but would still keep it for flares or when you are trying to debug an issue

**Pull request merged**